vivisect/analysis/amd64/instrhook.py

```diff
@@ -0,0 +1,50 @@
+"""
+Finds functions containing specific instructions to determine if
+their operands should be a  pointer locations.
+This module should be run after codeblocks analyis pass.
+"""
+
+import envi
+import vivisect.impemu.monitor as viv_imp_monitor
+
+verbose = False
+
+
+class instrhook_watcher (viv_imp_monitor.EmulationMonitor):
+
+    def __init__(self, vw, tryva):
+        viv_imp_monitor.EmulationMonitor.__init__(self)
+        self.vw = vw
+        self.tryva = tryva
+        self.hasret = False
+        self.mndist = {}
+        self.insn_count = 0
+        self.lastop = None
+        self.badcode = False
+        self.badops = vw.arch.archGetBadOps()
+
+    def prehook(self, emu, op, rip):
+        # Not sure if we need to stop emulation altogether,
+        # since we're only looking for specific instructions anyway?
+        if op in self.badops:
+            emu.stopEmu()
+            raise Exception("Hit known BADOP at 0x%.8x %s" % (rip, repr(op)))
+
+        if op.mnem == 'stosb' or op.mnem == 'stosd' or op.mnem == 'stosq':
```

Comment:
You should just be able to make a constant tuple at the top of this file:
`STOS = ('stosb', 'stosd', 'stosq')`

And then do:
`if op.mnem in STOS:`